classes/class-wc-connect-shipping-label.php

```diff
@@ -353,19 +353,26 @@ protected function get_form_data( WC_Order $order ) {
 
 		protected function get_states_map() {
 			$result = array();
-			foreach ( WC()->countries->get_countries() as $code => $name ) {
-				$result[ $code ] = array( 'name' => html_entity_decode( $name ) );
-			}
-			foreach ( WC()->countries->get_states() as $country => $states ) {
-				$result[ $country ]['states'] = array();
-				foreach ( $states as $code => $name ) {
-					if ( 'US' === $country && in_array( $code, $this->unsupported_states ) ) {
-						continue;
+			$all_countries = WC()->countries->get_countries();
+			$allowed_countries = array( 'US', 'PR' );
+
+			foreach ( $allowed_countries as $country_code ) {
+				$country_data = array( 'name' => html_entity_decode( $all_countries[ $country_code ] ) );
+				$states = WC()->countries->get_states( $country_code );
+
+				if ( $states ) {
+					$country_data['states'] = array();
+					foreach ( $states as $state_code => $name ) {
+						if ( 'US' === $country_code && in_array( $state_code, $this->unsupported_states ) ) {
```

Comment:
What do you think about using `$this->unsupported_states` for all countries and nesting what's currently there under a `US` key? It might be unlikely that we restrict states from other countries, but it removes some US-specific hardcoding.